ros1_ign_point_cloud/src/point_cloud.cc

```diff
@@ -0,0 +1,347 @@
+{
+  /// \brief Callback when the depth camera generates a new frame.
+  /// This is called in the rendering thread.
+  /// \param[in] _scan Depth image data
+  /// \param[in] _width Image width in pixels
+  /// \param[in] _height Image height in pixels
+  /// \param[in] _channels Number of channels in image.
+  /// \param[in] _format Image format as string.
+  public: void OnNewDepthFrame(const float *_scan,
+            unsigned int _width, unsigned int _height,
+            unsigned int _channels,
+            const std::string &_format);
+
+  /// \brief Get depth camera from rendering.
+  /// \param[in] _ecm Immutable reference to ECM.
+  public: void LoadDepthCamera(const ignition::gazebo::EntityComponentManager &_ecm);
+
+  /// \brief Get RGB camera from rendering.
+  /// \param[in] _ecm Immutable reference to ECM.
+  public: void LoadRgbCamera(const ignition::gazebo::EntityComponentManager &_ecm);
+
+  /// \brief Rendering scene which manages the cameras.
+  public: ignition::rendering::ScenePtr scene_;
+
+  /// \brief Entity ID for sensor within Gazebo.
+  public: ignition::gazebo::Entity entity_;
+
+  /// \brief Rendering depth camera
+  public: std::shared_ptr<ignition::rendering::DepthCamera> depth_camera_;
+
+  /// \brief Rendering RGB camera
+  public: std::shared_ptr<ignition::rendering::Camera> rgb_camera_;
+
+  /// \brief Keep latest image from RGB camera.
+  public: ignition::rendering::Image rgb_image_;
+
+  /// \brief Message populated with latest image from RGB camera.
+  public: sensor_msgs::Image rgb_image_msg_;
+
+  /// \brief Connection to depth frame event.
+  public: ignition::common::ConnectionPtr depth_connection_;
+
+  /// \brief Node to publish ROS messages.
+  public: std::unique_ptr<ros::NodeHandle> rosnode_;
+
+  /// \brief Point cloud ROS publisher.
+  public: ros::Publisher pc_pub_;
+
+  /// \brief Current simulation time.
+  public: std::chrono::steady_clock::duration current_time_;
+
+  /// \brief Frame ID to put in message header. Defaults to sensor scoped name.
+  public: std::string frame_id_;
+};
+
+//////////////////////////////////////////////////
+PointCloud::PointCloud() : dataPtr(std::make_unique<PointCloudPrivate>())
+{
+}
+
+//////////////////////////////////////////////////
+void PointCloud::Configure(const ignition::gazebo::Entity &_entity,
+    const std::shared_ptr<const sdf::Element> &_sdf,
+    ignition::gazebo::EntityComponentManager &_ecm,
+    ignition::gazebo::EventManager &)
+{
+  this->dataPtr->entity_ = _entity;
+
+  // Initialize ROS
+  if (!ros::isInitialized())
+  {
+    int argc = 0;
+    char** argv = NULL;
+    ros::init(argc, argv, "ignition", ros::init_options::NoSigintHandler);
+    ROS_INFO_NAMED("ros1_ign_point_cloud", "Initialized ROS");
+  }
+
+  // Sensor scoped name
+  auto scoped_name = ignition::gazebo::scopedName(this->dataPtr->entity_, _ecm, "/", false);
+
+  // ROS node
+  auto node_name = _sdf->Get<std::string>("node_name", scoped_name).first;
```

Comment:
This is not a big deal, but the node name is always "ignition" right now, which I think is being set on line 119. The `NodeHandle` constructor's argument is the namespace. Do we even need node_name?

Reply:
Ohhh it's a pretty big deal, the parameter is completely wrong :sweat_smile: I'm confusing the ROSes. Fixed in 7eede34757687e87f494d7450f7837be4142b09b